Fixes bundle mode with enabled sourceMapTraceback

Toliak opened the pull request:
See #1109 for details

Perryvw commented on src/transpilation/bundle.ts:
```diff
@@ -77,11 +77,11 @@ export function getBundleResult(program: ts.Program, files: ProcessedFile[]): [t
     const tableEntryHead = `[${modulePath}] = function() `;
     const tableEntryTail = " end,\n";
 
-    return joinSourceChunks([tableEntryHead, sourceMapNode ?? code, tableEntryTail]);
```
Seems like you're just throwing away the source map information of the file? Did you test this? Seems to me like this would just break the source map completely.

Toliak replied:
The code already contains source map information. I'll figure it out and write tests tomorrow.

Toliak replied:
I've changed the source map tracebacks behavior for bundles. The details are below

Perryvw reviewed (changes requested):
Would expect to see at least one test added to verify this actually fixes the problem.

Toliak commented:
## Main changes
- `__TS__originalTraceback`  is able to save two types of source maps.  
  **First type** is default: `Record<number, number>` lua line is being mapped into TS line.  
  **The second type** is: `Record<number, { line : number, file: string }`: lua line is mapped into TS line and source file name.
  The first type has been already applied to single file compilation. And I have applied the second type for bundle compilation. The reason -- in bundle mode runtime Lua should know about target file (because bundle filename != source file)
- I have added `printStackTraceBundleOverride` function into `bundle.ts`. This function generates the **second type** of source map.
- `LuaPrinter`'s method `printFile` will not generate `{#SourceMapTraceback}` in bundle mode anymore. Because Source Map Traceback should be constructed for the whole bundled file
- `getBundleResult` function inserts the traceback before the `entryPoint`

## Misc changes
- Moved `{#SourceMapTraceback}` magic-macro-string into a static field `sourceMapTracebackPlaceholder` of `LuaPrinter` class
- Haven't touched `sourceMapNode` in moduleSourceNode (because it is not necessary)
- Moved helpful `lineAndColumnOf` from `test\unit\printer\sourcemaps.spec.ts` into `test\unit\printer\utils.ts`

## Tests coverage
- Moved default bundle test into `bundle/bundle-two-files`
- Tested source map generation with `bundle/bundle-source-maps` test
- Both tests are located in `test\transpile\bundle.spec.ts` and wrapped into `describe`
